cloud_Azure/terraform/module/examples/multiple_accounts_multiple_resource_group/azure_cli.py

```diff
@@ -0,0 +1,46 @@
+import logging
+import os
+import time
+from typing import Any, Optional
+
+from az.cli import az
+
+log = logging.getLogger(__name__)
+
+
+def az_cli(command: str, max_attempts: int = 1) -> Optional[Any]:
+    """
+    Azure CLI commands issued in a quick succession may fail,
+    eg. when trying to configure a resource that is still being created
```

Comment:
The "Asynchronous operations" sections in [Azure CLI documentation](https://docs.microsoft.com/en-us/cli/azure/use-cli-effectively?view=azure-cli-latest) claims that the CLI internally waits for all asynchronous operations to complete. Are your observations indication of bugs in the CLI or do we need to call it differently to really make all operations synchronous?

Reply:
Also, is it even safe to retry arbitrary Azure CLI commands? I do realize that there are very few commands used in this specific context, but there is always a potential that this code may be later re-used in another project (with the assumption that the code has been used successfully for a while).

Reply:
my observation is: cli waits, but sometimes it is not long enough to be ready to accept subsequent commands that modify specified resource.
retry here is optional and default is no-retry. I think the user should know what it means to retry a command